src/app/labeling/qgsmaptoolmovelabel.cpp

```diff
@@ -182,16 +182,18 @@ void QgsMapToolMoveLabel::cadCanvasPressEvent( QgsMapMouseEvent *e )
         return;
       }
 
-      int xCol = -1, yCol = -1;
+      int xCol = -1, yCol = -1, pointCol = -1;
 
-      if ( !mCurrentLabel.pos.isDiagram && !labelMoveable( vlayer, mCurrentLabel.settings, xCol, yCol ) )
+      if ( !mCurrentLabel.pos.isDiagram && !labelMoveable( vlayer, mCurrentLabel.settings, xCol, yCol, pointCol ) )
       {
-        QgsPalIndexes indexes;
+        if ( mCurrentLabel.settings.dataDefinedProperties().isActive( QgsPalLayerSettings::PositionPoint ) )
+          mCurrentLabel.settings.dataDefinedProperties().property( QgsPalLayerSettings::PositionPoint ).setActive( false );
```

Comment:
why this change?

Reply:
When using the auxiliary layer we store the values as X and Y coordinates as it is now

Reply:
So is this situation even possible to hit? If so, I don't think we should just silently throw away the user's data defined configuration for PositionPoint. Better to show a message bar warning instead that the label cannot be moved because of .... and abort early here.

Reply:
Yes it is possible when the point position is defined as a read only expression (not pointing to a writable geometry column).
Ok it make sense to change it like you described.